### ISSUE TITLE
Dynamically Generate Alt-Text for Images

### ISSUE THREAD

**Issue by neodjandre2** [sybrew/the-seo-framework#180]
Is there a way to dynamically generate alt-text for images based on the post title ?

I used to use a plugin called SEO Friendly Images to do this but it hasn't been updated for years now..

Apparently Alt-Text for images is quite important for Google SEO.

Many thanks,
Andrew

**Comment by sybrew** [comment]
Hi Andrew / @neodjandre2 

Alt-text is important! But, we don't possess image interpretation (like Facebook and Google has).
I believe you mean this with "Dynamic", i.e. auto-generated?

In any case, when images are put on pages, Google interprets the subject bound to it. So that helps them get found.
Also, WordPress allows you to set alt-tags through the image editing interface. Simply applying an image Title will be sufficient.

**Comment by neodjandre2** [comment]
yes ideally we wouldn't want to spend time adding alt-text to images on upload. Maybe we will try another plugin then thanks. 

Although I did find this function which does that, but I am a bit sceptical of using it..

```
function add_alt_tags($content)
{
        global $post;
        preg_match_all('/<img (.*?)\/>/', $content, $images);
        if(!is_null($images))
        {
                foreach($images[1] as $index => $value)
                {
                        if(!preg_match('/alt=/', $value))
                        {
                                $new_img = str_replace('<img', '<img alt="'.$post->post_title.'"', $images[0][$index]);
                                $content = str_replace($images[0][$index], $new_img, $content);
                        }
                }
        }
        return $content;
}
add_filter('the_content', 'add_alt_tags', 99999);
```

**Comment by sybrew** [comment]
Please, don't use that 🙉 

It will label every image as the post title. It might resolve the missing alt-tags, but having none is better than forcing probable duplicates. 

I really can't give you any suggestions other than using interpretation software like [Facebook's Deepmask](https://github.com/facebookresearch/deepmask). But, that needs to be connected to a deep-learning database.

In any case, software like this isn't out there for WordPress. I'd suggest to manually add alt-tags (titles) or leave them empty for the time being.

**Comment by petworldglobal** [comment]
Is this now included in the suite? Rank Math has it in their free version...

**Comment by alexschomb** [comment]
Any update or timeline on this?

**Comment by sybrew** [comment]
Since this issue was opened, the world has changed a lot with generative AI. Completing this issue is now much more feasible, and many solutions that do this have spawned since. 

Still, I have no ETA for this feature. While waiting, check out these solutions:
1. https://wordpress.org/plugins/alttext-ai/
2. https://wordpress.org/plugins/ai-alt-text-generator/
3. https://wordpress.org/plugins/ai-image-alt-text-generator-for-wp/
